### ISSUE TITLE
Pad audio fragment

### ISSUE THREAD

**Issue by Alexey322** [jik876/hifi-gan#90]
Why do we need pad audio fragment while receiving its mel spec?

`y = torch.nn.functional.pad(y.unsqueeze(1), (int((n_fft-hop_size)/2), int((n_fft-hop_size)/2)), mode='reflect')`

**Comment by leminhnguyen** [comment]
Hi @Alexey322

I think the author used padding for doing `stft` (aka fast fourier transform) on all frames of the input audio segment.
```python
spec = torch.stft(y, n_fft, hop_length=hop_size, win_length=win_size, window=hann_window[str(y.device)],
                      center=center, pad_mode='reflect', normalized=False, onesided=True)
```
You can check the [torch.stft](https://pytorch.org/docs/stable/generated/torch.stft.html) function from the API doc for more details.

**Comment by Alexey322** [comment]
Hi @leminhnguyen.

Thanks for your reply. Why can't we just align the fragment size with convolutions? With v1 configuration 29 mels correspond to 8192 samples, what's the point of adding redundant data?
